### PR TITLE
Update James project SHA1

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

**Comment by chibenwa** [comment]
Distributed server fails to start:

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.173 s <<< FAILURE! - in com.linagora.tmail.james.app.EncryptedDistributedServerTest

[ERROR] com.linagora.tmail.james.app.EncryptedDistributedServerTest  Time elapsed: 0.173 s  <<< ERROR!

com.google.inject.CreationException: 

Unable to create injector, see the following errors:


1) [Guice/MissingImplementation]: No implementation for MailQueueView$Factory was bound.


Requested by:

1  : RabbitMQMailQueueFactory$PrivateFactory.<init>(RabbitMQMailQueueFactory.java:81)

      \_ for 7th parameter

     at RabbitMQMailQueueFactory.<init>(RabbitMQMailQueueFactory.java:130)

      \_ for 3rd parameter

     at RabbitMQModule.provideRabbitMQMailQueueFactoryProxy(RabbitMQModule.java:77)

      \_ for 1st parameter

     at RabbitMQModule.provideRabbitMQMailQueueFactoryProxy(RabbitMQModule.java:77)

      \_ installed by: Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$CombinedModule -> Modules$CombinedModule -> Modules$CombinedModule -> Modules$CombinedModule -> Modules$CombinedModule -> Modules$OverrideModule -> RabbitMQModule
```

We need this module chooser: 

```
GuiceJamesServer.forConfiguration(configuration)
            .combineWith(MODULES)
            .combineWith(MailQueueViewChoice.ModuleChooser.choose(configuration.getMailQueueViewChoice()))
```

The next issue you will get is the need to bind delegation store and authoriztor in Distributed james LDAP and Combined user repository.

File: tmail-backend/combined-identity/src/main/java/com/linagora/tmail/combined/identity/CombinedUsersRepositoryModule.java [hunk 1]
```diff
@@ -34,6 +38,8 @@ public void configure() {
 
         bind(UsersDAO.class).to(CombinedUserDAO.class);
         bind(UsersRepository.class).to(CombinedUsersRepository.class);
+        bind(DelegationStore.class).to(CassandraDelegationStore.class);
+        bind(Authorizator.class).to(DelegationStoreAuthorizator.class);
```
Comment (chibenwa) [inline tmail-backend/combined-identity/src/main/java/com/linagora/tmail/combined/identity/CombinedUsersRepositoryModule.java:42]: How about LDAP?

Comment (quantranhong1999) [reply tmail-backend/combined-identity/src/main/java/com/linagora/tmail/combined/identity/CombinedUsersRepositoryModule.java:42]: I thought that LDAP would use LdapUsersRepositoryModule (NaiveDelegationStore and UserRepositoryAuthorizator) same as James side?

**Comment by quantranhong1999** [comment]
```

[ERROR]   ElasticSearchIntegrationTest>AbstractMessageSearchIndexTest.searchWithPDFAttachmentShouldReturnMailsWhenAttachmentContentMatches:1395 

Expecting actual:

  []

to contain exactly (and in same order):

  [MessageUid{uid=4}]

but could not find the following elements:

  [MessageUid{uid=4}]
```

I confirm this test (mailbox-elasticsearch-v6) failed on local also, will have a look deeper tmr.

**Comment by Arsnael** [comment]
Read it

**Comment by vttranlina** [comment]
Read it

**Comment by quantranhong1999** [comment]
This test on master is failed also...

**Comment by chibenwa** [comment]
IMO @Disabled...

**Comment by quantranhong1999** [comment]
How I fix this:
Debug Search Hit of ES 7:
```json
"attachments" : [
      {
        "mediaType" : "application",
        "subtype" : "pdf",
        "fileName" : null,
        "fileExtension" : null,
        "contentDisposition" : "attachment",
        "textContent" : "The attachment contains a beautiful banana.\n\n\n"
      }
    ]
```

SearchHit of ES 6:
```json
"attachments" : [
      {
        "mediaType" : "application",
        "subtype" : "pdf",
        "fileName" : null,
        "fileExtension" : null,
        "contentDisposition" : "attachment",
        "textContent" : null
      }
    ]
```

=> Smt is wrong with ES6 text extractor.

I tried PDFTextExtractor instead of TikaTextExtractor -> test passed. 
=> Smt is wrong with ES6 TikaTextExtractor
=> I have a look at what changed regarding that on ES7 => Copy/Paste.

**Comment by quantranhong1999** [comment]
```
[ERROR] Failures: 

[ERROR]   MemoryTeamMailboxesTest.EmailSubmissionSetGetShouldAcceptTeamMailbox JSON documents are different:

Different keys found in node "methodResponses[0][1]", missing: "methodResponses[0][1].created", extra: "methodResponses[0][1].notCreated", expected: <{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","created":{"k1490":"${json-unit.ignore}"},"newState":"2c9f1b12-b35a-43e6-9af2-0106fb53a943"}> but was: <{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","newState":"2c9f1b12-b35a-43e6-9af2-0106fb53a943","notCreated":{"k1490":{"description":"block()/blockFirst()/blockLast() are blocking, which is not supported in thread reactor-http-epoll-8","type":"serverFail"}}}>
```
Likely another bug.

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
```
/home/jenkins/build/workspace/Tmail_build_PR-442/tmail-backend/integration-tests/jmap/memory-jmap-integration-tests/target/surefire-reports for the individual test results.

[ERROR]   MemoryTeamMailboxesTest.EmailSubmissionSetGetShouldAcceptTeamMailbox JSON documents are different:

Different keys found in node "methodResponses[0][1]", missing: "methodResponses[0][1].created", extra: "methodResponses[0][1].notCreated", expected: <{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","created":{"k1490":"${json-unit.ignore}"},"newState":"2c9f1b12-b35a-43e6-9af2-0106fb53a943"}> but was: <{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","newState":"2c9f1b12-b35a-43e6-9af2-0106fb53a943","notCreated":{"k1490":{"description":"block()/blockFirst()/blockLast() are blocking, which is not supported in thread reactor-http-epoll-8","type":"serverFail"}}}>

java.lang.IllegalStateException: block()/blockFirst()/blockLast() are blocking, which is not supported in thread reactor-http-epoll-8

	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:83)

	at reactor.core.publisher.Mono.block(Mono.java:1707)

	at reactor.core.scala.publisher.SMono.block(SMono.scala:108)

	at reactor.core.scala.publisher.SMono.block$(SMono.scala:107)

	at reactor.core.scala.publisher.ReactiveSMono.block(ReactiveSMono.scala:8)

	at com.linagora.tmail.team.TMailCanSendFrom.validTeamMailbox(TMailCanSendFrom.scala:25)

	at com.linagora.tmail.team.TMailCanSendFrom.userCanSendFrom(TMailCanSendFrom.scala:14)

	at org.apache.james.jmap.method.EmailSubmissionSetMethod.$anonfun$validate$3(EmailSubmissionSetMethod.scala:289)

	at org.apache.james.jmap.method.EmailSubmissionSetMethod.$anonfun$validate$3$adapted(EmailSubmissionSetMethod.scala:289)

	at scala.collection.immutable.List.filter(List.scala:515)

	at org.apache.james.jmap.method.EmailSubmissionSetMethod.validate(EmailSubmissionSetMethod.scala:289)

	at org.apache.james.jmap.method.EmailSubmissionSetMethod.$anonfun$sendEmail$7(EmailSubmissionSetMethod.scala:257)

	at reactor.core.scala.publisher.SMono$.$anonfun$fromTry$1(SMono.scala:1532)

	at reactor.core.scala.publisher.SMono$.$anonfun$fromTry$1$adapted(SMono.scala:1531)

	at reactor.core.scala.publisher.package$.$anonfun$scalaConsumer2JConsumer$2(package.scala:45)

	at reactor.core.scala.publisher.package$.$anonfun$scalaConsumer2JConsumer$2$adapted(package.scala:45)

	at scala.Option.foreach(Option.scala:437)

```

**Comment by quantranhong1999** [comment]
I am thinking about adding `.subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER)` back to `EmailSubmissionSetMethod`. But my IDE is crapping with jdk 17 packages, can not confirm whether it fix or not yet.

**Comment by chibenwa** [comment]
IMO we should try:

```
class TMailCanSendFrom @Inject()(rrt: RecipientRewriteTable, aliasReverseResolver: AliasReverseResolver, teamMailboxRepository: TeamMailboxRepository) extends CanSendFromImpl(rrt, aliasReverseResolver) {
  override def userCanSendFrom(connectedUser: Username, fromUser: Username): Boolean =
    super.userCanSendFrom(connectedUser, fromUser) || validTeamMailbox(connectedUser, fromUser)

  private def validTeamMailbox(connectedUser: Username, fromUser: Username): Boolean =
    TeamMailbox.asTeamMailbox(fromUser.asMailAddress()) match {
      case Some(teamMailbox) => SMono(teamMailboxRepository.listMembers(teamMailbox))
        .filter(connectedUser.equals(_))
        .hasElement
        .onErrorResume {
          case _: TeamMailboxNotFoundException => SMono.just(false)
          case e => SMono.error(e)
        }
        .block()
      case None => false
    }

  override def allValidFromAddressesForUser(user: Username): stream.Stream[MailAddress] = stream.Stream.concat(
    super.allValidFromAddressesForUser(user),
    Flux.from(teamMailboxRepository.listTeamMailboxes(user)).map(_.asMailAddress).collect(ImmutableList.toImmutableList).block().stream())
}
```

Previous version relies on toStream which blocks while consuming the events.

The version I propose will block while evaluating the pipeline, and is already managed...

File: tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TMailCanSendFrom.scala [hunk 2]
```diff
@@ -22,6 +22,8 @@ class TMailCanSendFrom @Inject()(rrt: RecipientRewriteTable, aliasReverseResolve
           case _: TeamMailboxNotFoundException => SMono.just(false)
           case e => SMono.error(e)
         }
+        .asJava()
+        .share()
```
Comment (chibenwa) [inline tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TMailCanSendFrom.scala:26]: Why do we need `.share` ? What is this operator doing?

Comment (quantranhong1999) [reply tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TMailCanSendFrom.scala:26]: It is some kind of caching the Mono/turn it to a hot Publisher.
rf: https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#share--
https://stackoverflow.com/questions/67534602/webtestclient-returning-illegalstateexception-block-blockfirst-blocklast

File: tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailboxRepository.scala [hunk 3]
```diff
@@ -190,6 +191,8 @@ class TeamMailboxRepositoryImpl @Inject()(mailboxManager: MailboxManager) extend
       .filter(entryKey => NameType.user.equals(entryKey.getNameType))
       .map(entryKey => Username.of(entryKey.getName))
       .distinct()
+      .asJava()
+      .subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER)
```
[the remark this answers is not in the document]
Comment (quantranhong1999) [reply tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailboxRepository.scala:195]: This likely would reactify `listRights` on James side. Would that be ok to you?

[the remark this answers is not in the document]
Comment (chibenwa) [reply tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailboxRepository.scala:195]: I mean `flatMap(x => SMono.fromCallable(xxx => listRights).subcribeOn(yyy))`

> This likely would reactify listRights on James side. Would that be ok to you?

:+1: but please don't require this to fix TMail build!

File: tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TMailCanSendFrom.scala (hunk 2 above)
Comment (chibenwa) [reply tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TMailCanSendFrom.scala:26]: Do we need to turn into a hot publisher just before blocking???

Comment (quantranhong1999) [reply tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TMailCanSendFrom.scala:26]: Not really. Actually, this .`share` use is dangerous. This cause `DistributedTeamMailboxesTest` sometimes fail sometimes green. Maybe we need to reactify `userCanSendFrom` at the end.

Comment (quantranhong1999) [reply tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TMailCanSendFrom.scala]: > reactify userCanSendFrom

Adapted a part for TMailCanSendFrom on TMail side and leveraged it in EmailSubmissionSetMethod. I will check where else to leverage in James and push a PR.

**Comment by Arsnael** [comment]
```
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.6.1:compile (scala-compile-first) on project team-mailboxes: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:4.6.1:compile failed: Compilation failed: InterfaceCompileFailed -> [Help 1]
```

**Comment by quantranhong1999** [comment]
Green.

**Pull request merged**